app/util/job.ts

```diff
@@ -24,29 +24,32 @@ async function cleanupWorkItemsForJobID(jobID: string, logger: Logger): Promise<
 }
 
 /**
-   * Set the state of a job to 'paused' unless already in a terminal state then save it.
+   * Pause a job and then save it.
    *
-   * @param tx - the transaction to perform the updates with
-   * @param job - the job to save and update
+   * @param jobID - the id of job (requestId in the db)
    * @param logger - the logger to use for logging errors/info
-   * @param message - the job's paused message
+   * @param username - the name of the user requesting the pause - null if the admin
    * @throws {@link ConflictError} if the job is already in a terminal state.
  */
 export async function pauseAndSaveJob(
-  tx: Transaction,
-  job: Job,
-  finalStatus: JobStatus,
+  jobID: string,
   logger: Logger,
-  message?,
+  username: string,
 ): Promise<void> {
-  job.pause(message);
-  try {
+  let job;
+  await db.transaction(async (tx) => {
```

Comment:
We do this exact lookup in three different places now. We should refactor this into a function.

Reply:
Done.